Initial pass for external ES (#483)
infrawatch/documentation: doc-Service-Telemetry-Framework/modules/con_primary-parameters-of-the-servicetelemetry-object.adoc

[discrete]
=== Enabling Elasticsearch as a storage back end for events

[NOTE]

> Older versions of {ProjectShort} would manage Elasticsearch objects for the community supported Elastic Cloud on Kuberneters Operator (ECK). This functionality is deprecated as of {ProjectShort} 1.5.3 and will be removed in a future version. Future versions will continue to support forwarding to an existing Elasticsearch instance (which can be one deployed by ECK), but will not manage the creation of Elasticsearch objects. When upgrading an {ProjectShort} deployment, any existing Elasticsearch object and deployment will remain intact, but will no longer be managed by {ProjectShort}.

Reviewer: ```suggestion
Older versions of {ProjectShort} would manage Elasticsearch objects for the community supported Elastic Cloud on Kuberneters Operator (ECK). This functionality is deprecated as of {ProjectShort} 1.5.3. Future versions of Service Telemetry Operator will continue to support forwarding to an existing Elasticsearch instance (which can be one deployed by ECK), but will not manage the creation of Elasticsearch objects. When upgrading an {ProjectShort} deployment, any existing Elasticsearch object and deployment will remain intact, but will no longer be managed by {ProjectShort}.
```

Reviewer: As part of documentation style, we don't promise anything in writing, so I've removed that bit, but otherwise this looks good thanks!

Reviewer: Also since this comment is a single line, I think you can just do `NOTE: <single_line_note>`. For admonition blocks like this, the content will usually get wrapped like so:

```
[NOTE]
====
Some sentence.

Another sentence.
====
```

This probably works fine in this case, but if someone adds another paragraph or something in the future it can cause generation problems you might not notice.

Author: I hadn't rendered it yet, so I'll just implement the multi-line version, otherwise I'll have to remember to change it in the future if we expand it. Thanks!

> To enable Elasticsearch as a storage back end for events, you must configure the `ServiceTelemetry` object.

Reviewer: ```suggestion
To enable events forwarding to Elasticsearch as a storage back end, you must configure the `ServiceTelemetry` object.
```

[discrete]
=== Enabling Elasticsearch as a storage back end for events

[NOTE]
====

> Older versions of {ProjectShort} would manage Elasticsearch objects for the community supported Elastic Cloud on Kuberneters Operator (ECK). This functionality is deprecated as of {ProjectShort} 1.5.3. Future versions of Service Telemetry Operator will continue to support forwarding to an existing Elasticsearch instance (which can be one deployed by ECK), but will not manage the creation of Elasticsearch objects. When upgrading an {ProjectShort} deployment, any existing Elasticsearch object and deployment will remain intact, but will no longer be managed by {ProjectShort}.

Reviewer: ```suggestion
Older versions of {ProjectShort} would manage Elasticsearch objects for the community supported Elastic Cloud on Kubernetes Operator (ECK). Elasticsearch management functionality is deprecated as of {ProjectShort} 1.5.3. Future versions of Service Telemetry Operator will continue to support forwarding to an existing Elasticsearch instance (which can be deployed and managed by ECK), but will not manage the creation of Elasticsearch objects. When upgrading an {ProjectShort} deployment, any existing Elasticsearch object and deployment will remain intact, but will no longer be managed by {ProjectShort}.
```

Reviewer: Kuberneters!

    events:
      elasticsearch:
        enabled: true
        forwarding:

>           hostUrl: https://external-elastic-http.domain:9200

Reviewer: I think it might be worth to have this value match what we would have deployed previously to make it simpler for someone to migrate from managed-ES to external-ES with their existing ECK-based ES deployment.

Author: Users who are migrating won't need to touch it though, or even really know about the parameter at all. The default value (which is shown elsewhere in these docs) suggests that the ES instance should be deployed in the same OCP namespace as STF; something I'd rather discourage. So I thought this example would better emphasize that this points to an externally managed ES, and not something that lives with STF.

          tlsServerName: ""
          tlsSecretName: elasticsearch-es-cert
          userSecretName: elasticsearch-es-elastic-user
          useBasicAuth: true
          useTls: true
----

> . Create the secret named in the `userSecretName`` parameter to store the basic auth credentials

Reviewer: ```suggestion
. Create the secret named in the `userSecretName` parameter to store the basic auth credentials
```

Author: See? I told you I hadn't rendered it yet ;)
